src/core/string_set.cc

```diff
@@ -112,6 +112,10 @@ unsigned StringSet::AddBatch(absl::Span<std::string_view> span, uint32_t ttl_sec
   return res;
 }
 
+StringSet::iterator StringSet::GetRandomMember(size_t search_offset) {
+  return iterator{DenseSet::GetRandomIterator(search_offset)};
```

Comment:
you could use a reservoir sampling. i.e. while the iterator is a link decide with 1/k probability whether to progress or break. this would cover items in the links as well. i.e. at first with 1/2 to progress, then 1/3 etc.